core/src/main/java/io/questdb/griffin/engine/functions/bind/CompiledFilterSymbolBindVariable.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.bind;
+
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.*;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.SymbolFunction;
+import io.questdb.griffin.engine.functions.constants.SymbolConstant;
+
+/**
+ * String bind variable function wrapper used in SQL JIT. Also used to handle deferred
+ * (unknown at compile time) symbol literals.
+ */
+public class CompiledFilterSymbolBindVariable extends SymbolFunction implements ScalarFunction {
```

Comment:
we should discuss why `IndexedParameterLinkFunction` cannot be used here

Reply:
As we discussed earlier, `CompiledFilterSymbolBindVariable` should stay as is since its main purpose is to map symbol char sequences to their int values.